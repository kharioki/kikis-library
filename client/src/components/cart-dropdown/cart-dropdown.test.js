import React from "react";
import { shallow } from "enzyme";

import CartDropdown from "./cart-dropdown.component";

describe("CartDropdown tests", () => {
  let wrapper;
  const mockCartItems = [{ id: 1 }, { id: 2 }, { id: 3 }];

  beforeEach(() => {
    wrapper = shallow(<CartDropdown cartItems={mockCartItems} />);
  });

  it("should render CartDropdown component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render cart empty message if cartItems is empty", () => {
    const mockCartItems = [];

    const newWrapper = shallow(<CartDropdown cartItems={mockCartItems} />);
    expect(newWrapper.exists(".empty-message")).toBe(true);
  });
});
