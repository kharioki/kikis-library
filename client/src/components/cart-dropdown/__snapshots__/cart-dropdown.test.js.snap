// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartDropdown tests should render CartDropdown component 1`] = `ShallowWrapper {}`;
